lib/commons/dom/is-visible.js

```diff
@@ -45,12 +46,16 @@ dom.isVisible = function(el, screenReader, recursed) {
 		el = el.host; // grab the host Node
 	}
 
-	style = window.getComputedStyle(el, null);
+	if (node && typeof node[cacheName] !== 'undefined') {
+		return node[cacheName];
```

Comment:
Can you add 2 tests:
1. that the cache is set after first accessing this method
2. that the cache is used when accessing this a second time

Reply:
I'm going to use this comment to group all the tests for caching comments. 

We already have code that tests that all of these functions are working as expected. Is there a particular reason you would like the unit tests to be tied to how the function was implemented internally? 

One of the nice things about these drastic code changes (especially for `getNodeFromTree`) is that I could completely swap out the internal implementation and I didn't have to update any of the tests. If we start adding tests that make sure a node is added to the cache, or retrieved from the cache, or that private cache properties are added, then we tie a lot of tests to the implementation of the internal code. We also force changes to that implementation to require changes to the unit tests.

IMO, we shouldn't test the internals of a function as the internals can easily change, creating fragile tests. We should only test that the function returns an expected set of outputs for a given set of inputs.

Reply:
Fair enough :)